### PR TITLE
Sustain 245-246: Added pipeline to call scraper service

### PR DESCRIPTION
**Pull request by rakhshankhan**
# JIRA Ticket Link

https://skaarkzo-capstone.atlassian.net/browse/SUSTAIN-245
https://skaarkzo-capstone.atlassian.net/browse/SUSTAIN-246

# Summary of Changes

- Added service logic to handle sending request to data scraper service.
- Added controller logic for LLM to call to retrieve scraped data.

### CODE REVIEW

File: app/service/search_service.py [hunk 1]
```diff
@@ -1,4 +1,5 @@
 from app.model.dto.search_dto import SearchDTO
 
+
 def process_search(input_text: str) -> SearchDTO:
     return SearchDTO(companyName=input_text)
```
Comment (skhan943) [inline app/service/search_service.py:5]: Can we standardize company_name

File: app/core/config.py [hunk 2]
```diff
@@ -8,6 +9,9 @@ class Settings(BaseSettings):
 
     SCRAPER_BASE_URL: str = "http://localhost:8001"
 
+    # TODO: Add a non-None timeout when data scraper handles ending process on no response received.
+    MAX_TIMEOUT: Timeout = Timeout(None)
```
Comment (skhan943) [inline app/core/config.py:13]: Please explain the timeout a bit more with a comment, thanks!

**Review by skhan943 — CHANGES_REQUESTED** [review]
Generally looks good, minor fixes

File: app/service/scraper_service.py [hunk 3]
```diff
@@ -1,36 +1,40 @@
...
 from app.core.config import settings
 
 
 class ScraperService:
 
     @staticmethod
-    async def get_company_scraped_data(company_name: str):
-        url = f"{settings.SCRAPER_BASE_URL}/api/scraper/company/{company_name}"
+    async def get_company_scraped_data(search_request: SearchRequest):
+        url = f"{settings.SCRAPER_BASE_URL}/api/scraper/company/"
```
Comment (zaidokal) [inline app/service/scraper_service.py:12]: Let's put all the URLs in the settings file? Including the /api/scraper/company. 

Could do something like this in the settings file:
export const API_ENDPOINTS = {
    SINGLE_COMPANY: `${API_BASE_URL}/company`,
    SCRAPER_API: `${API_BASE_URL}/api/scraper/company`
}

Comment (zaidokal) [inline app/service/scraper_service.py:12]: Also can we change the one to call the database? It is currently http://localhost:8000/api/main/companies, its too similar to /company

**Review by zaidokal — CHANGES_REQUESTED** [review]

**Review by zaidokal — APPROVED** [review]

**Review by skhan943 — APPROVED** [review]

**Pull request merged**